### PR TITLE
Jupyter-notebook for the community module

### PR DESCRIPTION
**Pull request by cndolo**
This PR adds a new jupyter-notebook for the `community` module. I made minor changes to the documentation too.

### CODE REVIEW

File: networkit/_NetworKit.pyx [hunk 1]
```diff
@@ -4939,6 +4939,14 @@ cdef extern from "<networkit/community/PLP.hpp>":
 cdef class PLP(CommunityDetector):
 	""" Parallel label propagation for community detection:
 	Moderate solution quality, very short time to solution.
+		Parameters
+		----------
+		G : networkit.Graph
+			The graph on which the algorithm has to run.
+		updateThreshold : integer
```
Comment (angriman) [inline networkit/_NetworKit.pyx]:
```suggestion
		updateThreshold : int
```

File: notebooks/Community.ipynb [hunk 2]
```diff
@@ -0,0 +1,221 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# NetworKit Community Detection"
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "# Community Detection with NetworKit"
```

File: notebooks/Community.ipynb [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we will cover some algorithms on the `community` module. Community detection is concerned with identifying groups of nodes which are significantly more densely connected to eachother than to the rest of the network. The first step is to import NetworKit."
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "In this notebook we will cover some community detection algorithms implemented in the `community` module of NetworKit. Community detection is concerned with identifying groups of nodes which are significantly more densely connected to each other than to the rest of the network.
    
As a first step we NetworKit:"
```

File: notebooks/Community.ipynb [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+   "outputs": [],
+   "source": [
+    "import networkit as nk"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `community` module provides a top-level function, [detectCommunities(G, algo=None, inspect=True)](https://networkit.github.io/dev-docs/python_api/community.html?highlight=detect#networkit.community.detectCommunities) to quickly perform community detection of a given with a suitable algorithm and print some statistics about the result. If we pass a graph to the this function, it choses a suitable algorithm and can be used as follows:"
```
Comment (angriman) [inline notebooks/Community.ipynb]: The first sentence is a bit confusing. Which algorithm does this function run? How is it chosen? Does quickly mean that the algorithm is fast?
The second sentence is redundant, you can simply write "This function can be used as follows:"

File: notebooks/Community.ipynb [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    "# Read graph\n",
+    "G = nk.readGraph(\"../input/karate.graph\", nk.Format.METIS)\n",
+    "communities = nk.community.detectCommunities(G)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following sections now cover two community detection algorithms, `PLM` and `PLM`, and will give you and you introduction on using them."
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "The following sections cover two popular community detection algorithms, `PLM` and `PLM`, and will illustrate how to use them."
```

File: notebooks/Community.ipynb [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+   "metadata": {},
+   "source": [
+    "NetworKit provides a parallel implementation of the well-known Louvain method, which can be found in the [PLM](https://networkit.github.io/dev-docs/python_api/community.html?highlight=plm#networkit.community.PLM) class. It yields a high-quality solution at reasonably fast running times. The constructor `(Graph, refine=False, gamma=0.1, par='balance', maxIter=32, turbo=True, recurse=True)` expects a [networkit.Graph](https://networkit.github.io/dev-docs/python_api/networkit.html?highlight=graph#networkit.Graph) as a mandatory parameter. The Boolean parameter `refine` adds a second move phase to refine the communities when set to true. The parameter `gamma` defines the multi-resolution modularity parameter. The string `par` defines the parallelization strategy. `maxIter` is the maximum number of iterations for move phase, and is set to 32 by default. When `turbo` is set to true, the algorithm is faster but uses O(n) additional memory per thread. Set `recurse`to true in order to use recursive coarsening. Refer to [this]( http://journals.aps.org/pre/abstract/10.1103/PhysRevE.89.049902) for more details on recursive coarsening.\n",
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "NetworKit provides a parallel implementation of the well-known Louvain method, which can be found in the [PLM](https://networkit.github.io/dev-docs/python_api/community.html?highlight=plm#networkit.community.PLM) class. It yields a high-quality solution at reasonably fast running times. The constructor `PLM(Graph, refine=False, gamma=0.1, par='balance', maxIter=32, turbo=True, recurse=True)` expects a [networkit.Graph](https://networkit.github.io/dev-docs/python_api/networkit.html?highlight=graph#networkit.Graph) as a mandatory parameter. If the parameter `refine` is set to true, the algorithm performs a second move phase to refine the communities. The parameter `gamma` defines the multi-resolution modularity parameter. The string `par` defines the openmp parallelization strategy. `maxIter` is the maximum number of iterations for move phase. When `turbo` is set to true, the algorithm is faster but uses O(n) additional memory per thread. Set `recurse`to true in order to use recursive coarsening. Refer to [this]( http://journals.aps.org/pre/abstract/10.1103/PhysRevE.89.049902) for more details on recursive coarsening.\n",
```

File: notebooks/Community.ipynb [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    "\n",
+    "For example if we want to run PLM with `refine` set to true and `gamma`= 0.02 while using the rest of the default parameters, we start by reading a graph from a file. Herafter, we can initialse the algorithm."
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "In the example below we run PLM with `refine` set to true and `gamma`= 0.02 while leaving the rest of the parameters to their default values."
```

File: notebooks/Community.ipynb [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read graph\n",
+    "G = nk.readGraph(\"../input/karate.graph\", nk.Format.METIS)"
```
Comment (angriman) [inline notebooks/Community.ipynb]: Remove this cell.

File: notebooks/Community.ipynb [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the [detectCommunities(G, algo=None, inspect=True)](https://networkit.github.io/dev-docs/python_api/community.html?highlight=detect#networkit.community.detectCommunities) function, we specify `PLM` as the algorithm of choice, and pass the parameters as follows: "
```
Comment (angriman) [inline notebooks/Community.ipynb]: Remove also this cell.

File: notebooks/Community.ipynb [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Choose and initialse algorithm \n",
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "# Read graph\n",
    "G = nk.readGraph(\"../input/karate.graph\", nk.Format.METIS)"
    "# Choose and initialse algorithm \n",
```

File: notebooks/Community.ipynb [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+    "plmCommunities = nk.community.detectCommunities(G, algo=nk.community.PLM(G, True,0.02))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The result of community detection is a partition of the node set into disjoint subsets. It is represented by the [Partition](https://networkit.github.io/dev-docs/python_api/networkit.html?highlight=partition#networkit.Partition) data strucure, which provides several methods for inspecting and manipulating a partition of a set of elements (which need not be the nodes of a graph)."
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "The output of the `detectCommunities` function is a partition of the nodes of the graph. It is represented by the [Partition](https://networkit.github.io/dev-docs/python_api/networkit.html?highlight=partition#networkit.Partition) data structure, which provides several methods for inspecting and manipulating a partition of a set of elements."
```

File: notebooks/Community.ipynb (hunk 10 above)
Comment (angriman) [reply notebooks/Community.ipynb]: It would be more illustrative if the output communities are more than one.

File: notebooks/Community.ipynb [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+   "outputs": [],
+   "source": [
+    "print(\"the biggest subset has size {0}\".format(max(plmCommunities.subsetSizes())))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The contents of a partition object can be written to file in a simple format, in which each line `i` contains the subset id of node `i`."
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "The contents of a partition object can be written to file in a simple format, in which the `i`-th line contains an integer representing the subset id of node `i`."
```

File: notebooks/Community.ipynb [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+   "metadata": {},
+   "source": [
+    "## PLP "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Label Propagation algorithm is an algorithm for finding communities in a graph. NetworKit provides a parallel implementation, [PLP(G, updateThreshold,maxIterations)](https://networkit.github.io/dev-docs/python_api/community.html?highlight=plp#networkit.community.PLP). The constructor expects a [networkit.Graph](https://networkit.github.io/dev-docs/python_api/networkit.html?highlight=graph#networkit.Graph) as a mandatory parameter. The parameter `updateThreshold` dictates the number of nodes that have to be changed in each iteration so that a new iteration starts, and `maxIterations` is the maximum number of iterations.\n",
```
Comment (angriman) [inline notebooks/Community.ipynb]: Write PLP default parameters.

Comment (cndolo) [reply notebooks/Community.ipynb]: The default values of both parameters is `none`, which is not in the Python documentation. How should I express this? Also, should I then I add then it to the documentation?

Comment (angriman) [reply notebooks/Community.ipynb]: Yes you can add it to the Python documentation and mention it in the notebook.

**Comment by cndolo** [comment]
I have addressed the comments you made @angriman; let me know if I should change anything else.

File: notebooks/Community.ipynb [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+    "In the example below we run PLM with `refine` set to true while leaving the rest of the parameters to their default values.\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Choose and initialse algorithm \n",
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "# Choose and initialize algorithm \n",
```

File: notebooks/Community.ipynb [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+    "G = nk.readGraph(\"../input/jazz.graph\", nk.Format.METIS)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Choose and initialse algorithm \n",
```
Comment (angriman) [inline notebooks/Community.ipynb]:
```suggestion
    "# Choose and initialize algorithm \n",
```

**Pull request merged**